src/modules/simulator/simulator_mavlink.cpp

```diff
@@ -589,6 +597,41 @@ void Simulator::send()
 			parameters_update(false);
 			poll_topics();
 			send_controls();
+			send_sim_tone();
+		}
+	}
+}
+
+void Simulator::send_sim_tone()
+{
+	static long tone_timer = 0;
+	long cur_time_in_ms;
+	struct timespec spec;
+
+	clock_gettime(CLOCK_MONOTONIC, &spec);
+
+	cur_time_in_ms = spec.tv_sec * 1000 + spec.tv_nsec / 1.0e6;
+
+	// don't grab the next tone until the current one is scheduled to be done
+
+	if (tone_timer < cur_time_in_ms) {
+
+		struct {
+			unsigned int frequency;
+			unsigned int duration;
+			unsigned int silence;
+		} tone_info = {};
+
+		if (_tunes.get_next_note(tone_info.frequency, tone_info.duration, tone_info.silence) > 0) {
```

Comment:
I'm not saying you must, but you could additionally check if the note you are going to send is with ` ... && tone_info.frequency > 0 && tone_info.duration > 0 && tone_info.strength > 0`. Then you won't send the pause notes.